### PR TITLE
Add support for multiple Lovelace dashboards

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

Add support for multiple Lovelace dashboards.

Requires https://github.com/home-assistant/home-assistant/pull/32134

To do:
- [x] verify it doesn't break demo and cast.
- [x] Migrate resources to be fetched from `lovelace/resources`
- [x] Preload resources on page startup

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
lovelace:
  dashboards:
    bla:
      mode: yaml
      filename: bla.yaml
      sidebar:  # Optional
        title: Make it rain
        icon: mdi:cash-100
```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/ha-panel-lovelace.ts [hunk 1]
```diff
@@ -130,10 +132,29 @@ class LovelacePanel extends LitElement {
...
-    subscribeLovelaceUpdates(this.hass!.connection, () =>
-      this._lovelaceChanged()
-    );
+    if (this.urlPath === null) {
+      // we don't want to unsub as we want to stay informed of updates
+      subscribeLovelaceUpdates(this.hass!.connection, this.urlPath, () =>
+        this._lovelaceChanged()
+      );
+    } else if (!resourcesLoaded) {
+      // We only load resources from default config.
```
Comment (bramkragten) [inline src/panels/lovelace/ha-panel-lovelace.ts]: I don't think this is necessary, as we handle duplicate resources in `loadLovelaceResources`, and besides that, we don't load new resources now when the config is changed.

**Review by bramkragten — APPROVED** [review]

File: src/panels/lovelace/hui-root.ts [hunk 2]
```diff
@@ -497,6 +498,12 @@ class HUIRoot extends LitElement {
         | undefined;
 
       if (!oldLovelace || oldLovelace.config !== this.lovelace!.config) {
+        if (this.lovelace!.config.resources) {
```
Comment (balloob) [inline src/panels/lovelace/hui-root.ts]: We can't just restore this, we only want to load resources from the default Lovelace config, not from the other dashboards.

File: src/entrypoints/core.ts [hunk 3]
```diff
@@ -89,13 +93,10 @@ window.hassConnection.then(({ conn }) => {
...
   subscribeUser(conn, noop);
 
-  if (location.pathname === "/" || location.pathname.startsWith("/lovelace/")) {
-    (window as WindowWithLovelaceProm).llConfProm = fetchConfig(
-      conn,
-      null,
-      false
-    );
-  }
+  (window as WindowWithLovelaceProm).llConfProm = fetchResourcesAndConfig(
```
Comment (balloob) [inline src/entrypoints/core.ts]: We shouldn't do this in 1 function. This should be 2 functions, as they always should run in parallel. I want to start processing either the resources or the config while the other has not returned yet. By waiting for both, we're now stalling for no reason.

File: src/data/lovelace.ts [hunk 4]
```diff
@@ -117,6 +130,17 @@ export const fetchConfig = (
     url_path: urlPath,
     force,
   });
+export const fetchResourcesAndConfig = async (
```
Comment (balloob) [inline src/data/lovelace.ts]: We should remove this function.

File: src/panels/lovelace/ha-panel-lovelace.ts [hunk 5]
```diff
@@ -284,11 +282,13 @@ class LovelacePanel extends LitElement {
       }
     }
 
-    this._state = "loaded";
-    this._setLovelaceConfig(conf, confMode);
-    if (resources) {
+    if (resourceProm) {
+      const resources = await resourceProm;
```
Comment (balloob) [inline src/panels/lovelace/ha-panel-lovelace.ts]: Why should we wait for resources to be loaded? It's fine to load those in later too? We can already start showing some UI.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**